### PR TITLE
Feature/82 day of week group control

### PR DESCRIPTION
**Pull request by ScottBTR**
DayOfWeek Enum --> CalendarDayOfWeek Enum Bitflag.

- Works with quick select dropdown for days of week
- Altered iOS to work with new bitflag enum

### CODE REVIEW

File: Samples/Samples/ViewModel/CalendarEventAddViewModel.cs [hunk 1]
```diff
@@ -8,6 +8,7 @@
 using System.Windows.Input;
 using Xamarin.Essentials;
 using Xamarin.Forms;
+using PropertyChangingEventArgs = System.ComponentModel.PropertyChangingEventArgs;
```
Comment (nickrandolph) [inline Samples/Samples/ViewModel/CalendarEventAddViewModel.cs]: why is this necessary? can't we just have
using System.ComponentModel

File: Samples/Samples/ViewModel/CalendarEventAddViewModel.cs [hunk 2]
```diff
@@ -322,22 +376,74 @@ public bool IsMonthDaySpecific
...
+            if (!IsUpdatingCheckBoxGroup)
+            {
+                OnPropertyChanged(nameof(IsNone));
+                OnPropertyChanged(nameof(IsWeekdays));
+                OnPropertyChanged(nameof(IsWeekend));
+                OnPropertyChanged(nameof(IsAllDays));
+            }
+        }
+
+        ObservableCollection<CalendarDayOfWeekSwitch> recurrenceDays;
```
Comment (nickrandolph) [inline Samples/Samples/ViewModel/CalendarEventAddViewModel.cs]: Is this in the correct order in the file? 
Should this be readonly now?

File: Samples/Samples/ViewModel/CalendarEventAddViewModel.cs (hunk 1 above)
Comment (ScottBTR) [reply Samples/Samples/ViewModel/CalendarEventAddViewModel.cs]: Not required anymore will remove

File: Samples/Samples/ViewModel/CalendarEventAddViewModel.cs [hunk 3]
```diff
@@ -322,22 +376,74 @@ public bool IsMonthDaySpecific
...
-            new DayOfTheWeekSwitch() { Day = DayOfTheWeek.Sunday }
-        };
+            try
+            {
+                IsUpdatingCheckBoxGroup = true;
+                var currentVal = bitFlagValue;
+                var maxValue = (int)CalendarDayOfWeek.Saturday;
+                for (var i = maxValue; i > 0; i /= 2)
+                {
+                    RecurrenceDays[(int)Math.Log(i, 2)].IsChecked = currentVal >= i;
```
Comment (nickrandolph) [inline Samples/Samples/ViewModel/CalendarEventAddViewModel.cs]: Hmmm not sure we should be relying on >= here for this.
Wouldn't it be better to cycle through the RecurrenceDays array and set value based on logical & with bitFlagValue
eg
RecurrenceDays[i].IsChecked=(RecurrenceDays[i].Day & bitFlagValue)>0

File: Samples/Samples/ViewModel/CalendarEventAddViewModel.cs (hunk 2 above)
Comment (ScottBTR) [reply Samples/Samples/ViewModel/CalendarEventAddViewModel.cs]: I'm not sure if it can be readonly, as if I try to make it readonly it wants OnChildCheckBoxChangedEvent to be static, which doesn't let me call OnPropertyChanged

File: Samples/Samples/ViewModel/CalendarEventAddViewModel.cs (hunk 3 above)
Comment (ScottBTR) [reply Samples/Samples/ViewModel/CalendarEventAddViewModel.cs]: umm, I can't use just 'i' as the array value because that's currently being used as a bit flag in a sense, i.e. this for loops 'i' goes 64 > 32 > 16> 8 > 4> 2 > 1, That's why I've got the Math.Log.

Comment (nickrandolph) [reply Samples/Samples/ViewModel/CalendarEventAddViewModel.cs]: right but you don't need to  - you can just do a foreach in the RecurrenceDays array can't you?

File: Samples/Samples/ViewModel/CalendarEventAddViewModel.cs (hunk 2 above)
Comment (nickrandolph) [reply Samples/Samples/ViewModel/CalendarEventAddViewModel.cs]: Hmmm that shouldn't be the case - you're assuming that a readonly field has to be set inline - readonly field can be set in the constructor like you are doing now I think. If not, change to readonly property

Comment (ScottBTR) [reply Samples/Samples/ViewModel/CalendarEventAddViewModel.cs]: I see, yeah funny that, when I'm trying to understand all this complex features of MVVM I'm forgetting the basics of a constructor can be used to set readonly fields

File: Samples/Samples/ViewModel/CalendarEventAddViewModel.cs (hunk 3 above)
Comment (ScottBTR) [reply Samples/Samples/ViewModel/CalendarEventAddViewModel.cs]: Altered to use HasFlag as it's a much cleaner check

**Review by nickrandolph — APPROVED** [review]
